src/queue/index.js

```diff
@@ -0,0 +1,164 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+import * as Error from '../helpers/errors';
+import amqp from 'amqplib';
+import {isNotDefined} from '../helpers/utils';
+import log from '../helpers/logger';
+
+
+/**
+ * QUEUE_NAME - Queue name for importing data.
+ * @type {string} QUEUE_NAME
+ **/
+export const QUEUE_NAME = 'bookbrainz-import';
+
+/**
+ * Connection - Object containing functions to create and close RMQ connections.
+ * @type {Object} Connection
+ * @property {function} connect - Returns a connection object promise
+ * @property {function} shutdown - Takes in a connection promise and closes the
+ * 		connection
+ **/
+export const Connection = {
+	connect: function connect() {
+		let connection = null;
+		try {
+			connection = amqp.connect('amqp://localhost');
+		}
+		catch (err) {
+			Error.raiseError(Error.CONNECTION_ERROR)(err);
+		}
+		return connection;
+	},
+	shutdown: function shutdown(connectionPromise) {
+		if (isNotDefined(connectionPromise)) {
+			Error.undefinedValue('Connection.shutdown:: connectionPromise');
+			return null;
+		}
+
+		return connectionPromise
+			.then(connection => connection.close())
+			.catch(Error.raiseError(Error.CONNECTION_CLOSE_ERROR));
+	}
+};
+
+/** Class representing a RMQ Queue. */
+export class Queue {
+	/**
+     * Create a channel promise
+     * @param {Promise} connectionPromise - Connection promise
+     */
+	constructor(connectionPromise) {
+		if (isNotDefined(connectionPromise)) {
+			Error.undefinedValue('Queue.constructor:: connectionPromise');
+			return null;
+		}
+		this.channelPromise = connectionPromise
+			.then(conn => conn.createChannel())
+			.catch(Error.raiseError(Error.CHANNEL_ERROR));
+	}
+
+	/**
+     * Push a message into the queue
+     * @param {string} msg - The string containing message to be pushed into RMQ
+     */
+	push(msg) {
+		if (isNotDefined(msg)) {
+			log.error('Invalid message! Skipping.');
+			return;
+		}
+
+		const serializedMessage = JSON.stringify(msg);
+
+		if (isNotDefined(this.channelPromise)) {
+			Error.undefinedValue('Queue.push:: undefined channel.');
+		}
+
+		this.channelPromise.then(channel => {
+			if (isNotDefined(channel)) {
+				Error.undefinedValue(
+					'Queue.push:: Unable to get channel from promise.'
+				);
+			}
+			channel.assertQueue(QUEUE_NAME, {durable: true})
+				.then(ok => {
+					if (isNotDefined(ok)) {
+						Error.undefinedValue(
+							'Queue.push:: Could not assert queue.'
+						);
+					}
+
+					try {
+						channel.sendToQueue(
+							QUEUE_NAME,
+							new Buffer(serializedMessage),
```

Comment:
`new Buffer(...)` is deprecated - see https://nodejs.org/api/buffer.html#buffer_new_buffer_string_encoding - please switch to the appropriate replacement